app/js/lib/speech-recogniser.js

```diff
@@ -0,0 +1,54 @@
+'use strict';
+
+const p = Object.freeze({
+  isListening: Symbol('isListening'),
+  recognition: Symbol('recognition'),
+  supportsRecognition: Symbol('supportsRecognition'),
+});
+
+export default class SpeechRecogniser {
+  constructor() {
+    this[p.isListening] = false;
+
+    const Recognition =
+      window.SpeechRecognition || window.webkitSpeechRecognition;
+
+    this[p.supportsRecognition] = !!Recognition;
+
+    if (Recognition) {
+      this[p.recognition] = new Recognition();
+    } else {
+      this[p.recognition] = {};
+    }
```

Comment:
In the work we've done previously on Fxbox, we sealed the classes that are not extended. This enforces a better code hygiene by making sure that all properties are set in the constructor and that the code using it is not going to extend it.
This is done simply by adding `Object.seal(this);` as the last instruction in the constructor.
What do you think?